Error with npm v9+: `npm access 2fa-required is not a valid access command`

<!--- Provide a short summary of the issue in the title above -->

## Description

<!-- Write a clear and concise description of what the issue is -->

Publishing a package for the first time causes an error with `npm access`:

```bash
$ npm --version
9.3.1
$ np --version
7.7.0
$ np --yolo

...

  ✔ Prerequisite check
  ✔ Git
  ✔ Bumping version using npm
  ✔ Publishing package using npm
  ✖ Enabling two-factor authentication
  ✖ Enabling two-factor authentication
    → npm ERR!     /Users/k/.npm/_logs/2023-04-11T16_44_52_318Z-debug-0.log
    Pushing tags
    Creating release draft on GitHub

✖ Command failed with exit code 1: npm access 2fa-required <pkg-name> --otp <otp value>
npm ERR! code EUSAGE
npm ERR! 
npm ERR! 2fa-required is not a valid access command
npm ERR! 
npm ERR! Set access level on published packages
npm ERR! 
npm ERR! Usage:
npm ERR! npm access list packages [<user>|<scope>|<scope:team> [<package>]
npm ERR! npm access list collaborators [<package> [<user>]]
npm ERR! npm access get status [<package>]
npm ERR! npm access set status=public|private [<package>]
npm ERR! npm access set mfa=none|publish|automation [<package>]
npm ERR! npm access grant <read-only|read-write> <scope:team> [<package>]
npm ERR! npm access revoke <scope:team> [<package>]
npm ERR! 
npm ERR! Options:
npm ERR! [--json] [--otp <otp>] [--registry <registry>]
npm ERR! 
npm ERR! Run "npm help access" for more info
```

`npm access 2fa-required` was a command in `npm@8`:

<img width="1113" alt="Screenshot 2023-04-11 at 18 52 27" src="https://user-images.githubusercontent.com/1935696/231236379-52225a00-9189-4730-a350-af7c57e71d7d.png">


The `npm access 2fa-required` command seems to be gone from the `npm@9` docs:

<img width="1062" alt="Screenshot 2023-04-11 at 18 52 41" src="https://user-images.githubusercontent.com/1935696/231236277-2d017efd-2fd5-41d5-9efd-0d8495ea9d76.png">


## Steps to reproduce

<!--- Let us how how we can reproduce the issue on our end -->

1. Publish a package for the first time using `npm` v9+
2. 💥 See the error

## Expected behavior

<!--- Tell us what you expected to happen -->

No errors when publishing a package for the first time with `npm` v9+

## Environment

<!-- Mention which versions of np, Node.js, npm and Git you're using, as well as your OS and version -->

np - 7.7.0
Node.js - 18.14.1
npm - 9.3.1
Git - 2.39.2
OS - macOS 13.3.1

## Workaround

Downgrade `npm` to v8:

```bash
npm install --global npm@^8
```

cc @DerTimonius in case you want to do another PR :)

--- a/source/npm/enable-2fa.js
+++ b/source/npm/enable-2fa.js
@@ -2,10 +2,10 @@ import {execa} from 'execa';
 import {from, catchError} from 'rxjs';
 import semver from 'semver';
 import handleNpmError from './handle-npm-error.js';
-import {version} from './util.js';
+import {version as npmVersionCheck} from './util.js';
 
 export const getEnable2faArgs = async (packageName, options) => {
-	const npmVersion = await version();
+	const npmVersion = await npmVersionCheck();
 	const args = semver.satisfies(npmVersion, '>=9.0.0') ? ['access', 'set', 'mfa=publish', packageName] : ['access', '2fa-required', packageName];
 
 	if (options && options.otp) {

--- a/source/npm/enable-2fa.js
+++ b/source/npm/enable-2fa.js
@@ -6,7 +6,7 @@ import {version} from './util.js';
 
 export const getEnable2faArgs = async (packageName, options) => {
 	const npmVersion = await version();
-	const args = semver.satisfies(npmVersion, '>=9.0.0') ? ['access', 'set', 'mfa', 'publish', packageName] : ['access', '2fa-required', packageName];
+	const args = semver.satisfies(npmVersion, '>=9.0.0') ? ['access', 'set', 'mfa=publish', packageName] : ['access', '2fa-required', packageName];
 
 	if (options && options.otp) {
 		args.push('--otp', options.otp);

--- a/source/npm/enable-2fa.js
+++ b/source/npm/enable-2fa.js
@@ -1,9 +1,12 @@
 import {execa} from 'execa';
 import {from, catchError} from 'rxjs';
+import semver from 'semver';
 import handleNpmError from './handle-npm-error.js';
+import {version} from './util.js';
 
-export const getEnable2faArgs = (packageName, options) => {
-	const args = ['access', '2fa-required', packageName];
+export const getEnable2faArgs = async (packageName, options) => {
+	const npmVersion = await version();
+	const args = semver.satisfies(npmVersion, '>=9.0.0') ? ['access', 'set', 'mfa', 'publish', packageName] : ['access', '2fa-required', packageName];
 
 	if (options && options.otp) {
 		args.push('--otp', options.otp);

